prometheus_client/openmetrics/parser.py

```diff
@@ -406,37 +502,38 @@ def do_checks():
     for s in samples:
         suffix = s.name[len(name):]
         g = _group_for_sample(s, name, 'histogram')
-        if g != group or s.timestamp != timestamp:
-            if group is not None:
-                do_checks()
-            count = None
-            bucket = None
-            has_negative_buckets = False
-            has_sum = False
-            has_gsum = False
-            has_negative_gsum = False
-            value = 0
-        group = g
-        timestamp = s.timestamp
-
-        if suffix == '_bucket':
-            b = float(s.labels['le'])
-            if b < 0:
-                has_negative_buckets = True
-            if bucket is not None and b <= bucket:
-                raise ValueError("Buckets out of order: " + name)
-            if s.value < value:
-                raise ValueError("Bucket values out of order: " + name)
-            bucket = b
-            value = s.value
-        elif suffix in ['_count', '_gcount']:
-            count = s.value
-        elif suffix in ['_sum']:
-            has_sum = True
-        elif suffix in ['_gsum']:
-            has_gsum = True
-            if s.value < 0:
-                has_negative_gsum = True
+        if len(suffix) != 0:
```

Comment:
Rather than nesting everything further in what about a `continue` if `len(suffix) is 0`?